### ISSUE TITLE
Prerequisite knowledge for adding support for Docstrings

### ISSUE THREAD

**Issue by dyba** [codegram/gherkin-ruby#24]
I noticed this library doesn't yet support docstrings. Now, if I were to go and try to implement this feature, what would I need to know to do that?

I noticed there's a .y file and a .rex file. It looks like I would have to understand yacc. What about the .rex file? Does anyone have a link they can post where I can learn more about the syntax in that file? Is this the correct repo for learning more about rex: [https://github.com/tenderlove/racc](https://github.com/tenderlove/racc)?

I'm going to refer to issue #20 since @kalupa also wanted support for this feature.

**Comment by dyba** [comment]
Answering my own question: To learn more about rex, go here [https://github.com/tenderlove/rexical](https://github.com/tenderlove/rexical)

**Comment by txus** [comment]
Docstrings seem like a good thing to support. However, with the rewrite of Cucumber 2.0, I'm wondering whether there can be a unified effort to get a sane Gherkin parser working, or if @mattwynne already made some steps in that direction. Is Cucumber 2.0 using still the Ragel-based gherkin parser?

**Comment by mattwynne** [comment]
Yes we still use the same parser and will continue to do so when Gherkin 3 is released. I had no idea about this project, but I'd suggest it would be better to put effort into helping with Gherkin 3 instead.

**Comment by txus** [comment]
Fair enough. @dyba if you feel like implementing Docstrings, that would be awesome! As waiting for Gherkin 3 might be some time :)
